src/pages/api/account/register.ts

```diff
@@ -1,32 +1,35 @@
+import axios from "axios";
 import { NextApiRequest, NextApiResponse } from "next";
-import { API_URL } from "../../../config/index";
+import { config } from "../../../config/index";
 
 export default async (req: NextApiRequest, res: NextApiResponse) => {
   if (req.method === "POST") {
-    const { first_name, last_name, email, password, userRole } = req.body;
+    const { email, username, name, password, roles } = req.body;
 
     const body = JSON.stringify({
-      first_name,
-      last_name,
       email,
+      username,
+      name,
       password,
-      userRole,
+      roles,
     });
 
     try {
-      const apiRes = await fetch(`${API_URL}/api/user/`, {
+      const apiRes = await axios.post(`${config.ITEZ_API_URI}/api/users/`, {
```

Comment:
potential blocker here
This picks up `ITEZ_APPI_URI` from env vars meanwhile in there we have what is called `NEXT_API_URL`
so lets unify everything to `ITEZ_API_URI` and all should be streamlined properly